controllers/bsl.go

```diff
@@ -145,6 +156,23 @@ func (r *DPAReconciler) ReconcileBackupStorageLocations(log logr.Logger) (bool,
 	return true, nil
 }
 
+func (r *DPAReconciler) UpdateLabels(secretName string, namespace string, dpaName string) (bool, error) {
+	var secret corev1.Secret
+	secret, err := r.getProviderSecret(secretName)
+	if err != nil {
+		return false, err
+	}
+	if secret.Name != "" {
```

Comment:
Perhaps this should error out if `secret.Name == ""`? That would imply we didn't find the secret right? Or maybe not even error but just return with no err?

Reply:
it would mean that get provider secret failed in a very unexpected way right? I think we can assume that this case is covered in that function ?

Reply:
good catch..on it right now

Reply:
returning an error is secret is not found